api/src/main/java/com/github/retrooper/packetevents/protocol/world/BlockFace.java

```diff
@@ -153,6 +153,32 @@ public BlockFace getCW() {
         }
     }
 
+    /** Returns int representing cardinal directions with south as the starting index at 0. UP and DOWN return -1
+     * <p>
+     * Used by Mojang for calculations involving mob AI, decorations, paintings, item frames, maps,
+     * structure generation, campfires, stairs, connecting blocks, and flowerbeds.
+     * Also used for getting Directions and Blockfaces as Rotations.
+     *
+     * @return int horizontal id
+     */
+    public int getHorizontalID(BlockFace facing) {
+        switch (facing) {
```

Comment:
could you also remove the parameter and replace "facing" with "this"?

Reply:
Done